core/controllers/base_test.py

```diff
@@ -274,6 +288,132 @@ def test_renders_error_page_with_iframed(self):
             'Uh-oh! The Oppia exploration you requested may have been removed '
             'or deleted.', response.body)
 
+    def test_dev_mode_cannot_be_true_on_production(self):
+        del sys.modules['feconf']
+        server_software_swap = self.swap(
+            os, 'environ', {'SERVER_SOFTWARE': 'Production'})
+        assert_raises_regexp_context_manager = self.assertRaisesRegexp(
+            Exception, 'DEV_MODE can\'t be true on production.')
+        with assert_raises_regexp_context_manager, server_software_swap:
+            import feconf  # pylint: disable-all
+
+    def test_valid_pillow_path(self):
+        import appengine_config
+        assert_raises_regexp_context_manager = self.assertRaisesRegexp(
+            Exception, 'Invalid path for oppia_tools library: invalid_path')
+
+        def mock_os_path_join_for_pillow(*args):
+            path = ''
+            if args[1] == 'Pillow-6.0.0':
+                return 'invalid_path'
+            else:
+                for arg in args:
+                    path += arg + '/'
+                return path
+
+        pil_path_swap = self.swap(os.path, 'join', mock_os_path_join_for_pillow)
+        del sys.modules['appengine_config']
+
+        with assert_raises_regexp_context_manager, pil_path_swap:
+            import appengine_config  # pylint: disable-all
+
+    def test_valid_third_party_library_path(self):
+        import appengine_config
+        assert_raises_regexp_context_manager = self.assertRaisesRegexp(
+            Exception, 'Invalid path for third_party library: invalid_path')
+
+        def mock_os_path_join_for_pillow(*args):
```

Comment:
Why is there pillow-specific stuff here for a test called test_valid_third_party_library_path? It seems like a copy of the previous test, too.

Reply:
Done

Reply:
Please answer the question -- don't just say "Done".

Reply:
Yes that was a mistake. Removed it!